aztk/config.py

```diff
@@ -45,13 +50,20 @@ def _merge_dict(self, secrets_config):
             self.batch_account_name = batch.get('batchaccountname')
             self.batch_account_key = batch.get('batchaccountkey')
             self.batch_service_url = batch.get('batchserviceurl')
+            self.batch_resource_url = batch.get('resourceurl')
 
         storage = secrets_config.get('storage')
         if storage:
             self.storage_account_name = storage.get('storageaccountname')
             self.storage_account_key = storage.get('storageaccountkey')
             self.storage_account_suffix = storage.get('storageaccountsuffix')
 
+        serviceprincipal = secrets_config.get('serviceprincipal')
```

Comment:
Probably worth warning a user that one auth method will be used if more than 1 is added to this file. For example, if SharedKey && ServicePrincipal, then use ServicePrincipal.

Reply:
The checks ([here](https://github.com/Azure/aztk/pull/240/files#diff-c64e530c75ccd0def3ecc50fbf949b2fR33)) currently use the presence of the resource url to decide whether to use shared key or service principal auth:
- If `resourceurl` is present, require that all `serviceprincipal` keys are present, and shared key account name and key must be absent
- if `resourceurl` is absent, ignore the `serviceprincipal` section (because in the future it may be used by other things) and require shared key details to be present

So it is not possible to specify both shared key and service principal auth for Batch (it will fail with `Both Batch resource url and account name are set in secrets.yaml config, can use either shared secret or service principal login, not both`.

I think this is clearer than allowing both to be specified, but then ignoring the shared key credentials.

Although if resource url and account name are merged into one key, we'd need to come up with something else anyway as this logic would no longer be possible.

Reply:
This is a good point. I agree with you.